numpy/__init__.py

```diff
@@ -155,6 +155,15 @@
     from . import matrixlib as _mat
     from .matrixlib import *
 
+    # Deprecations introduced in 1.25.0, 2023-05-xx TODO update day
+    from . import compat
```

Comment:
This will generate a warning. Hence I think you don't want to do this import (`import numpy` should be warning-free), but leave it to `__getattr__` which already can make it available as a lazy import (which will then emit the warning from `compat/__init__.py` on access).

Reply:
Sure! I moved `compat` import to `__getattr__` method.